llms/llm.go

```diff
@@ -388,6 +388,7 @@ func (l *LLM) runToolCall(ctx context.Context, toolbox *tools.Toolbox, toolCall
 
 	return Message{
 		Role:       "tool",
+		Name:       toolCall.Name,
```

Comment:
Hmm this is not the purpose of this field and can break in rare cases. I think most providers have since moved away from the idea that they'd have multiple names for different assistants in the same message history, but I think we should try to steer clear of it anyway.

Maybe we can introduce `ToolCallName` instead if there's no other way to propagate the name?